[sca/capture] Add Ctrl-C handler to capture script

wettermo opened the pull request:
In case a huge number of traces is captured, this handler allows the
user, when capturing takes too much time, to abort the capture without
losing previously recorded data.
To use this handler it has to be registered within each of the capture
functions, where a Chipwhisperer project is used to store traces.

vogelpi commented on cw/capture.py:
```diff
@@ -17,12 +20,12 @@
+from cw_segmented import CwSegmented
+from pyXKCP import pyxkcp
 from util import device, plot
+from waverunner import WaveRunner
```
Question is this change needed due to lint or is it a remainder from the rebasing maybe?

wettermo replied:
I had a few problems with isort on my machine, which apparently kind of messed up the imports. I only realized that after I opened the PR, so I had to manually fix it.

vogelpi commented on cw/capture.py:
```diff
@@ -258,6 +274,7 @@ def capture_aes_random_batch(ot, ktp, capture_cfg, scope_type):
     num_segments_storage = 1
     # cw and waverunner scopes are supported fot batch capture.
     scope = SCOPE_FACTORY[scope_type](ot, capture_cfg)
+
```
I think this additioan newline (and the other two below) aren't required. I don't mind them being there to improve readability but then they should be added with the next commit which touches the surroundings of these lines anyway.

wettermo replied:
I see, but I don't understand which next commit you mean.

wettermo replied:
Ah sorry, got it!

vogelpi reviewed (commented):
Thanks @wettermo , this looks mostly good.

I have some nits, see comments and two more things:
1. Should we also register the handler inside the `capture_loop` function? This would also enable the feature in non-batch mode for AES, SHA3, KMAC.
2. The commit title of the first commit message is too long, it wraps.

wettermo commented:
Thanks @vogelpi for the review!
Good point to also register the handler inside the capture_loop function. I'll do that and I'll also reforge the commits to get rid of the linting/isort issues.

wettermo commented:
@vogelpi I cleaned up the commits, added the registering of the handler in the capture_loop, and put them together into one commit. Now it should look cleaner.

vogelpi reviewed (approved):
Thanks for addressing all comments @wettermo , this looks good to me!
